dotCMS/src/main/java/com/dotcms/saml/DotIdentityProviderConfigurationImpl.java

```diff
@@ -0,0 +1,130 @@
+package com.dotcms.saml;
+
+import com.dotcms.security.apps.AppSecrets;
+import com.dotcms.security.apps.AppsAPI;
+import com.dotcms.security.apps.Secret;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+
+import java.util.Optional;
+
+/**
+ * Default implementation to retrieve the configuration from apps
+ * @author jsanca
+ */
+public class DotIdentityProviderConfigurationImpl implements IdentityProviderConfiguration {
+
+    private final AppsAPI    appsAPI;
+    private final Host       host;
+    private final AppSecrets appSecrets;
+
+    public DotIdentityProviderConfigurationImpl(final AppsAPI appsAPI, final Host host) throws DotSecurityException, DotDataException {
+
+        this.appsAPI    = appsAPI;
+        this.host       = host;
+        this.appSecrets = this.appsAPI.getSecrets(DotSamlProxyFactory.SAML_APP_CONFIG_KEY,
+                true, host, APILocator.systemUser()).get();
+
+    }
+
+    private Optional<Secret> findSecret (final String key) {
+
+        return this.appSecrets.getSecrets().containsKey(key)?
+                Optional.ofNullable(this.appSecrets.getSecrets().get(key)) : Optional.empty();
+    }
+
+    @Override
+    public boolean isEnabled() {
+
+        final String enableKey           = SamlName.DOT_SAML_ENABLE.getPropertyName();
+        final Optional<Secret> secretOpt = this.findSecret(enableKey);
+        return secretOpt.isPresent()? secretOpt.get().getBoolean(): false;
```

Comment:
Codacy found an issue: [Ternary operators that can be simplified with || or &&](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)